includes/multi-currency/MultiCurrency.php

```diff
@@ -1441,28 +1441,56 @@ public function is_multi_currency_settings_page(): bool {
 		);
 	}
 
+	/**
+	 * Function used to compute the customer used currencies, used as internal callable for get_all_customer_currencies function.
+	 *
+	 * @return array
+	 */
+	public function callable_get_customer_currencies() {
+		global $wpdb;
+
+		$currencies  = $this->get_available_currencies();
+		$query_union = [];
+
+		if ( class_exists( 'Automattic\WooCommerce\Utilities\OrderUtil' ) &&
+					\Automattic\WooCommerce\Utilities\OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			foreach ( $currencies as $currency ) {
+				$query_union[] = sprintf(
+					'SELECT "%1$s" AS currency_code, EXISTS(SELECT currency FROM %2$s WHERE currency="%1$s" LIMIT 1) AS exists_in_orders',
+					$currency->code,
+					$wpdb->prefix . 'wc_orders'
+				);
+			}
+		} else {
+			foreach ( $currencies as $currency ) {
+				$query_union[] = sprintf(
+					'SELECT "%1$s" AS currency_code, EXISTS(SELECT meta_value FROM %2$s WHERE meta_key="_order_currency" AND meta_value="%1$s" LIMIT 1) AS exists_in_orders',
+					$currency->code,
+					$wpdb->postmeta
+				);
+			}
+		}
+
+		$sub_query  = join( ' UNION ALL ', $query_union );
+		$query      = "SELECT currency_code FROM ( $sub_query ) as subquery WHERE subquery.exists_in_orders=1 ORDER BY currency_code ASC";
+		$currencies = $wpdb->get_col( $query ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
```

Comment:
Might be a good idea to use caching, since this is still an expensive function to operate
https://developer.wordpress.org/reference/functions/wp_cache_set/

If set, cache can be cleared on save in the settings.

Reply:
not sure if we want to add another caching layer, we already cache the results on this method in `database_cache->get_or_add`